zebra-consensus/src/transaction/tests.rs

```diff
@@ -2577,6 +2563,83 @@ fn v5_with_duplicate_orchard_action() {
     });
 }
 
+/// Checks that the tx verifier handles consensus branch ids in V5 txs correctly.
+#[tokio::test]
+async fn v5_consensus_branch_ids() {
+    let state = service_fn(|_| async { unreachable!("state service should not be called") });
+
+    for network in Network::iter() {
+        let verifier = Buffer::new(Verifier::new_for_tests(&network, state), 10);
+
+        let (input, output, known_utxos) = mock_transparent_transfer(
+            Height(1),
+            true,
+            0,
+            Amount::try_from(1).expect("valid amount"),
+        );
+
+        let known_utxos = Arc::new(known_utxos);
+
+        let verify = |tx, nu: NetworkUpgrade| {
+            verifier
+                .clone()
+                .oneshot(Request::Block {
```

Comment:
Optional/Nitpick: Could we also check that the tx verifier returns the same error for `Request::Mempool` calls?

Reply:
The `Request::Mempool` request doesn't take any known UTXOs, so I had trouble composing a TX that would pass the validation. Do you have an idea of how we could construct one?

Reply:
We could use the mock state service, though it's not necessary if you'd like to resolve this and merge the PR.

Reply:
Done in [e970244](https://github.com/ZcashFoundation/zebra/pull/9063/commits/e97024448042e99bf72c067e280dfff686c2c988).